README.md

```diff
@@ -44,22 +44,25 @@ $ python3 -m oci_discovery.ref_engine_discovery -l debug example.com/app#1.0 2>/
   "example.com/app#1.0": {
     "roots": [
       {
-        "annotations": {
-          "org.opencontainers.image.ref.name": "1.0"
+        "root": {
+          "annotations": {
+            "org.opencontainers.image.ref.name": "1.0"
+          },
+          "casEngines": [
+            {
+              "protocol": "oci-cas-template-v1",
+              "uri": "https://a.example.com/cas/{algorithm}/{encoded:2}/{encoded}"
+            }
+          ],
+          "digest": "sha256:e9770a03fbdccdd4632895151a93f9af58bbe2c91fdfaaf73160648d250e6ec3",
+          "mediaType": "application/vnd.oci.image.manifest.v1+json",
+          "platform": {
+            "architecture": "ppc64le",
+            "os": "linux"
+          },
+          "size": 799
         },
-        "casEngines": [
-          {
-            "protocol": "oci-cas-template-v1",
-            "uri": "https://a.example.com/cas/{algorithm}/{encoded:2}/{encoded}"
-          }
-        ],
-        "digest": "sha256:e9770a03fbdccdd4632895151a93f9af58bbe2c91fdfaaf73160648d250e6ec3",
-        "mediaType": "application/vnd.oci.image.manifest.v1+json",
-        "platform": {
-          "architecture": "ppc64le",
-          "os": "linux"
-        },
-        "size": 799
+        "uri": "http://example.com/oci-index/app"
```

Comment:
Here needn't `protocol` entry? Maybe both `uri` and `protocol` are all necessary.

Reply:
> Here needn't `protocol` entry?

This isn't an engine config, it's the location from which this `root` was extracted.  It would be passed into a subsequent CAS-engine initialization as `base`, although in this case

 `.["example.com/app#1.0"].roots[0].root.casEngines[0].uri` is a full URI (not a reference), so the base URI won't be needed.  It helps with things like:

```json
{
  "example.com/app#1.0": {
    "roots": [
      {
        "root": {
          ...,
          "casEngines": [
            {
              "protocol": "oci-cas-template-v1",
              "uri": "../cas/{algorithm}/{encoded:2}/{encoded}"
            }
          ]
        },
        "uri": "http://example.com/oci-index/app"
      }
    ]
  }
}
```

So I don't think we need `protocol` here; just the base URI(s) for resolving any references in the returned root(s).